### DISCUSSION TITLE
Does Uno compile with VS2022 mac M2.

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#12291] [General]
I tried to make the simple template uno project on my M2 mac.  Using VS2022 Mac and i get the following error:

Target UnoResourcesGenerateOverrides:
.nuget/packages/uno.dsp.tasks/1.0.1/build/Uno.Dsp.Tasks.targets(20,3): error MSB4216: Could not run the "GenerateXamlTask" task because MSBuild could not create or connect to a task host with runtime "NET" and architecture "arm64".  Please ensure that (1) the requested runtime and/or architecture are available on the machine, and (2) that the required executable "MSBuild.exe" exists and can be run.
.nuget/packages/uno.dsp.tasks/1.0.1/build/Uno.Dsp.Tasks.targets(21,4): error MSB4027: The "GenerateXamlTask" task generated invalid items from the "OutputXamlFiles" output parameter. This object is no longer valid because the MetadataLoadContext that created it has been disposed.
    Done building target "UnoResourcesGenerateOverrides" in project. -- FAILED.

Here is my VS Environment
Visual Studio Enterprise 2022 for Mac Preview
Version 17.6 Preview (17.6 build 1496)
Installation UUID: 257ca543-81fa-4403-9660-7b920b8b011d

Runtime
.NET 7.0.3 (64-bit)
Architecture: Arm64
Microsoft.macOS.Sdk 13.1.1007; git-rev-head:8afca776a0a96613dfb7200e0917bb57f9ed5583; git-branch:release/7.0.1xx-xcode14.2

Roslyn (Language Service)
4.6.0-3.23180.6+99e956e42697a6dd886d1e12478ea2b27cceacfa

NuGet
Version: 6.4.0.117

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.203/Sdks
SDK Versions:
	7.0.203
	7.0.202
	7.0.200
	7.0.103
	7.0.102
	7.0.101
	6.0.408
	6.0.407
	6.0.406
	6.0.405
	6.0.404
MSBuild SDKs: /Applications/Visual Studio (Preview).app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	7.0.5
	7.0.4
	7.0.3
	7.0.2
	7.0.1
	6.0.16
	6.0.15
	6.0.14
	6.0.13
	6.0.12

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	3.1.32
	3.1.31
	3.1.30

Xamarin.Profiler
Version: 1.8.0.49
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode: 14.3 21812
Build: 14E222b

Xamarin.Mac
Version: 9.1.0.5 Visual Studio Enterprise
Hash: 7738c90c9
Branch: xcode14.2
Build date: 2023-01-25 15:56:14-0500

Xamarin.iOS
Version: 16.2.0.5 Visual Studio Enterprise
Hash: 7738c90c9
Branch: xcode14.2
Build date: 2023-01-25 15:56:15-0500

Xamarin Designer
Version: 17.6.3.9
Hash: 2648399ae8
Branch: remotes/origin/d17-6
Build date: 2023-04-27 00:18:22 UTC

Xamarin.Android
Version: 13.2.0.0 (Visual Studio Enterprise)
Commit: xamarin-android/d17-5/797e2e1
Android SDK: /Users/curtisconner/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		12.0 (API level 31)
		11.0 (API level 30)
		10.0 (API level 29)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 33.0.3
SDK Build Tools Version: 32.0.0

Build Information: 
Mono: 6dd9def
Java.Interop: xamarin/java.interop/main@149d70fe
SQLite: xamarin/sqlite/3.40.0@fdc1e34
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@9f02d77

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.16.1
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.6.0.50
Hash: a715dca
Branch: HEAD
Build date: 2023-04-27 00:18:27 UTC

Android Device Manager
Version: 0.0.0.1309
Hash: 06e3e77
Branch: HEAD
Build date: 2023-04-27 00:18:27 UTC

Build Information
Release ID: 1706001496
Git revision: 7673a20cf2d7409180fbdc86d3fd9fc7cc17a1e9
Build date: 2023-04-27 00:16:36+00
Build branch: release-17.6
Build lane: release-17.6

Operating System
Mac OS X 13.3.1
Darwin 22.4.0 Darwin Kernel Version 22.4.0
    Mon Mar  6 21:01:02 PST 2023
    root:xnu-8796.101.5~3/RELEASE_ARM64_T8112 arm64

**Comment by jeromelaban** [comment]
Thanks for the report. This is an issue we've been unable to reproduce reliably. Could you provide a [binlog of the build](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) ?

**Reply by cconner100** [reply]
So attached is a diagnostic build from vs2022 mac.  I could not do all the instructions in your link because you cant install a vs package to the new arm versions yet.  Sorry but if you have anything else i can do let me know
[Build.error.txt.zip](https://github.com/unoplatform/uno/files/11487534/Build.error.txt.zip)

**Comment by carldebilly** [comment]
@cconner100 A new pre-release version [`1.1.0-dev.14`](https://www.nuget.org/packages/Uno.Dsp.Tasks/1.1.0-dev.14) has been published on nuget. May you try it?

**Reply by cconner100** [reply]
That seems to work, was able to compile my mac catalyst project.  Thanks.
